packages/sdk/src/multicall.ts

```diff
@@ -18,29 +18,29 @@ export type EncodedRequest = {
   callData: string;
 };
 
-interface State {
+export interface State<MulticallType> {
   requests: Request[];
-  multicallClient: multicall.Instance;
+  multicallClient: MulticallType;
 }
 
 // Multicall class that exposes public functions to the user and recursively chains itself.  Acts immutable
 // if you store reference to the parent intsance. Children will contain mutated state.
-class Multicall implements State {
+export class Multicall<MulticallType extends multicall.Instance> implements State<multicall.Instance> {
```

Comment:
Why does this now need to be exported?

Why does it need to be a generic?

Could it just be
```ts
export class Multicall implements State<multicall.Instance> {
  public requests: Request[];
  public multicallClient: multicall.Instance;
  constructor(state: State<multicall.Instance>) {
```

Now, I _could_ see this making sense if you wanted to move most of this logic to a parent class and only implement the read call in the derived class. But that complexity probably isn't worth it.

Reply:
Okay I think this makes sense! Sorry for misunderstanding the code originally!

Reply:
because not only the state depends on the multicall Instance type, but the multicallClient attribute too :(